auth/store.go

```diff
@@ -943,12 +943,8 @@ func NewAuthStore(be backend.Backend, tp TokenProvider) *authStore {
 }
 
 func hasRootRole(u *authpb.User) bool {
```

Comment:
add a comment on the func. it is required that roles are sorted. i am not convinced this would improve perf though. but i guess it is simple enough to merge.

Reply:
OK, I'll add the comment.

Reply:
is this func covered by test somehow? i am a little concerned with the sorted assumption. people might accidentally break it.

Reply:
ok, I'll add tests for it.

Reply:
added the test, could you take a look when you have a time?